lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Stefan Heitmueller <stefan.heitmueller@gmx.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: docker_swarm
+    plugin_type: inventory
+    authors:
+      - Stefan Heitmüller (stefan.heitmueller@gmx.com)
+    short_description: Ansible dynamic inventory plugin for Docker swarm nodes.
+    requirements:
+        - python >= 2.7
+        - Docker SDK for Python > 1.10.0
+    description:
+        - Reads inventories from the Docker swarm API.
+        - Uses a YAML configuration file docker_swarm.[yml|yaml].
+    options:
+        plugin:
+            description: The name of this plugin, it should always be set to 'docker_swarm' for this plugin to recognize it as it's own.
+            type: str
+            required: true
+            choices: docker_swarm
+        host:
+            description: Socket of a Docker swarm manager node (tcp,unix).
+            type: str
+            required: true
+        verbose_output:
+            description: Toggle to (not) include all available nodes metadata (e.g. Platform, Architecture, OS, EngineVersion)
+            type: bool
+            default: yes
+        tls:
+            description: Connect using TLS without verifying the authenticity of the Docker host server.
+            type: bool
+            default: no
+        tls_verify:
+            description: Connect using TLS without verifying the authenticity of the Docker host server.
```

Comment:
we normally call this option `validate_certs`

Reply:
Actually, there's a bug in the description here, this option does *enable* cert checking (and TLS):
```suggestion
            description: Connect using TLS and verify the authenticity of the Docker host server.
```
The TLS options are named the same as the options for the docker daemon (whose names are similar to the names used when configuring Docker CLI/Python SDK). So let's stick to `tls_verify` for now, but fix the description.